src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -60,15 +84,68 @@ export class CourseExercisesComponent implements OnInit, OnDestroy {
         this.paramSubscription.unsubscribe();
     }
 
-    public groupExercises(selectedOrder: number): void {
+    get filterType() {
+        return ExerciseFilter;
+    }
+
+    /**
+     * Return the total number of exercises for the logged in user
+     */
+    getNrOfExercises(): number {
+        return sum([...this.exerciseCountMap.values()]);
+    }
+
+    /**
+     * Reorders all displayed exercises
+     */
+    flipOrder() {
+        this.sortingOrder = this.sortingOrder === this.ASC ? this.DESC : this.ASC;
+        localStorage.setItem(SortFilterStorageKey.ORDER, this.sortingOrder.toString());
+        this.applyFiltersAndOrder();
+    }
+
+    /**
+     * Filters all displayed exercises by applying the selected activeFilters
+     * @param filters The filters which should be applied
+     * @param active Should the activeFilters be active or inactive
+     */
+    filterUpdate(filters: ExerciseFilter[]) {
+        filters.forEach(filter => (this.activeFilters.has(filter) ? this.activeFilters.delete(filter) : this.activeFilters.add(filter)));
+        localStorage.setItem(SortFilterStorageKey.FILTER, Array.from(this.activeFilters).join(','));
+        this.applyFiltersAndOrder();
+    }
+
+    /**
+     * Checks if the given exercise still needs work, i.e. is not graded with 100%, or wasn't even started, yet.
+     * @param exercise The exercise which should get checked
+     */
+    private needsWork(exercise: Exercise): boolean {
+        return (
+            exercise.participations.some(participation => participation.results && participation.results.some(result => result.score !== 100)) ||
+            exercise.participations.every(participation => participation.results.length === 0)
+        );
+    }
+
+    /**
+     * Applies all selected activeFilters and orders and groups the user's exercises
+     */
+    private applyFiltersAndOrder() {
+        const filtered = this.course!.exercises.filter(
+            exercise =>
```

Comment:
Also here please use multiple named variables so that the different criteria can be understood.

Reply:
I now use some named variables